uPortal-webapp/src/main/resources/org/apereo/portal/groups/smartldap/init.crn

```diff
@@ -42,28 +43,75 @@
             newChain.addAll(chain);
             newChain.push(record.getGroup().getLocalKey());
             ScriptAttributes.RESPONSE.setAttribute('groupChain', newChain);
-            
-            // Next transform 'resolveDn' into 'resolveDnList' if the first is 
-            // present but not the later.  'resolveDn' came before 'resolveDnList,' 
-            // and older configs might still be using it;  this code should only 
+
+            // Next transform 'resolveDn' into 'resolveDnList' if the first is
+            // present but not the later.  'resolveDn' came before 'resolveDnList,'
+            // and older configs might still be using it;  this code should only
             // trigger on the first pass.
             if (req.hasAttribute('resolveDn') &amp;&amp; !req.hasAttribute('resolveDnList')) {
                 def resolveDnList = [];
                 resolveDnList.add(resolveDn);
                 ScriptAttributes.RESPONSE.setAttribute('resolveDnList', resolveDnList);
             }
-        
-            // Finally attempt to add this one to the collection of records.  The 
-            // 'isNew' flag indicates whether we have seen this group before;  
-            // if we have, there's no need to follow its tree of decendents. 
-            boolean isNew = GROUPS.add(record);
+
+            // Finally attempt to add this one to the collection of records.  The
+            // 'isNew' flag indicates whether we have seen this group before;
+            // if we have, there's no need to follow its tree of decendents.
+            boolean isNew = !GROUPS.contains(record);
             ScriptAttributes.RESPONSE.setAttribute('isNew', isNew);
+            if (isNew) {
+
+            // ESUP Modifications
+            //
+            //    The children are processed in order to
+            //    	- extract the idpath of the groups from their dn
+            //    	- remove the entries that are not groups (eg users).
+
+	        def newChildren=[];
+	        if (req.hasAttribute('childGroupKeyRegex')) {
+                def pattern = req.hasAttribute('pattern') ? req.getAttribute('pattern') : java.util.regex.Pattern.compile(ScriptAttributes.REQUEST.getAttribute("childGroupKeyRegex"));
+		        ScriptAttributes.RESPONSE.setAttribute('pattern', pattern);
+		            for (String child : record.getKeysOfChildren()) {
+		                def matcher = pattern.matcher(child);
+		                if (matcher.matches()) {
+			                newChildren.add(matcher.group(1));
+		                }
+                    }
+		            record = new LdapRecord(record.getGroup(), newChildren);
+	            }
+	            GROUPS.add(record);
+
+	            //   Parses the group names in order to handle the Grouper folders.
+
+                String current = record.getGroup().getName();
+                int sepPos;
+                boolean stop = false;
+	            boolean isGroup = true;
```

Comment:
Indented different than line 88?

Reply:
the indent is not well managed with crn files ...